pkg/controller/minio.go

```diff
@@ -179,16 +193,42 @@ func (c *Controller) checkOperatorCaForTenant(ctx context.Context, tenant *minio
 		}
 	}
 
-	tenantCaCert = tenantCaSecret.Data["public.crt"]
-	if !bytes.Equal(tenantCaCert, operatorCaCert) {
-		tenantCaSecret.Data["public.crt"] = operatorCaCert
+	publicCert, ok := tenantCaSecret.Data["public.crt"]
+	if ok && !bytes.Equal(publicCert, operatorPublicCert) {
+		tenantCaSecret.Data["public.crt"] = operatorPublicCert
 		_, err = c.kubeClientSet.CoreV1().Secrets(tenant.Namespace).Update(ctx, tenantCaSecret, metav1.UpdateOptions{})
 		if err != nil {
 			return false, err
 		}
+		// Reload certificates
+		c.getTransport(true)
 		return false, fmt.Errorf("'public.crt' in '%s/%s' secret changed, updating '%s/%s' secret", miniov2.GetNSFromFile(), OperatorCATLSSecretName, tenant.Namespace, OperatorCATLSSecretName)
 	}
 
+	tlsCert, ok := tenantCaSecret.Data["tls.crt"]
+	if ok && !bytes.Equal(tlsCert, operatorTLSCert) {
+		tenantCaSecret.Data["tls.crt"] = tlsCert
+		_, err = c.kubeClientSet.CoreV1().Secrets(tenant.Namespace).Update(ctx, tenantCaSecret, metav1.UpdateOptions{})
+		if err != nil {
+			return false, err
+		}
+		// Reload certificates
+		c.getTransport(true)
+		return false, fmt.Errorf("'tls.crt' in '%s/%s' secret changed, updating '%s/%s' secret", miniov2.GetNSFromFile(), OperatorCATLSSecretName, tenant.Namespace, OperatorCATLSSecretName)
```

Comment:
since we have this duplicated below almost exactly the same, we can abstract this to reduce duplicate code.

Reply:
That's ok. I will use the new feature to make `CreateOrUpdate` code shorter for everywhere.

Reply:
well, looking at it again, makes no sense to update several times the secret, refactored to update once instead. I don't think the remaining lines are not worth to have in a separated function.